addon/utils/should-add-token.js

```diff
@@ -2,9 +2,31 @@ export default function shouldAddToken (url, portalId = '') {
   // if this is a hosted service that belongs to the user's organization
   // if this is a service with stored credentials
   const id = portalId.toLowerCase();
-  return (
-    !!url.toLowerCase().match('arcgis.com/arcgis/rest/services') ||
-    !!url.toLowerCase().match(`arcgis.com/${id}/arcgis/rest`) ||
-    /utility\w+\.arcgis\.com\/sharing\/servers\/\w{32}\/rest\/services\/Hosted/i.test(url)
-  );
+  if (location && location.host && hostsMatch(location.host, url)) {
```

Comment:
I'm not wild about using `location` in a utility - ideally we'd pass this in... but it's hard to know where to draw that line - this is nice in that it's encapsulated, but - it's also impossible to test this part of the code. I'll let you make the call